conf/appoptics-init.sh

```diff
@@ -5,9 +5,54 @@ set -e
 CONFIG_FILE='/opt/appoptics/etc/config.yaml'
 TMP_FILE='/opt/appoptics/etc/config.yaml.tmp'
 
-if [ -n "$APPOPTICS_NODE_NAME" ]; then
-    sed "s/# hostname_alias: myhostname/hostname_alias: $APPOPTICS_NODE_NAME/" $CONFIG_FILE > $TMP_FILE
+# APPOPTICS_TOKEN is required
+if [ -n "${APPOPTICS_TOKEN}" ]; then
+    sed "s/APPOPTICS_TOKEN/${APPOPTICS_TOKEN}/" $CONFIG_FILE > $TMP_FILE
     cp $TMP_FILE $CONFIG_FILE
+else
+    echo "Please set APPOPTICS_TOKEN."
+    exit 1
 fi
 
+# Translate LOG_LEVEL to snapteld log_level 1-5
+if [ -n "${LOG_LEVEL}" ]; then
+    shopt -s nocasematch           # turn on case-insensitive matching for case stmt
+    case $LOG_LEVEL in
+        debug) loglevel=1;;
+        info) loglevel=2;;
+        warn|warning) loglevel=3;;
+        error) loglevel=4;;
+        fatal) loglevel=5;;
+        *) loglevel=3;;
+    esac
+    shopt -u nocasematch
+
+    sed "s/^#log_level: 3/log_level: ${loglevel}/" $CONFIG_FILE > $TMP_FILE
+    cp $TMP_FILE $CONFIG_FILE
+fi
+
+# Use APPOPTICS_HOSTNAME as hostname_alias
+if [ -n "$APPOPTICS_HOSTNAME" ]; then
+    sed "s/# hostname_alias: myhostname/hostname_alias: ${APPOPTICS_HOSTNAME}/" $CONFIG_FILE > $TMP_FILE
+    cp $TMP_FILE $CONFIG_FILE
+fi
+
+# Set to true to enable or disable specific plugins
+if [ "$APPOPTICS_ENABLE_DOCKER" = "true" ]; then
+    mv /tmp/appoptics-configs/docker.yaml /opt/appoptics/etc/plugins.d/docker.yaml
+fi
+
+if [ "$APPOPTICS_ENABLE_KUBERNETES" = "true" ]; then
+    mv /tmp/appoptics-configs/kubernetes.yaml /opt/appoptics/etc/plugins.d/kubernetes.yaml
+fi
+
+if [ "$APPOPTICS_DISABLE_HOSTAGENT" = "true" ]; then
```

Comment:
Why would someone want to disable the host agent functions ? I guess there is no harm in having it - just curious where will it be used

Reply:
Well, in our case for the deployment, we only need the snap agent to get k8s metrics. The rest would be duplicative of the daemonset.

Reply:
👍 sounds good for now - we can revisit later if needed